diff --git a/src/core/dbus/generator.cpp b/src/core/dbus/generator.cpp
index 74322bb..8674506 100644
--- a/src/core/dbus/generator.cpp
+++ b/src/core/dbus/generator.cpp
@@ -311,6 +311,8 @@ public:
     {
     }
 
+    Interface(const Interface&) = default;
+
     Interface& operator=(const Interface& rhs)
     {
         d = rhs.d;
@@ -362,6 +364,8 @@ public:
     {
     }
 
+    Method(const Method&) = default;
+
     Method& operator=(const Method& rhs)
     {
         d = rhs.d;
@@ -421,6 +425,8 @@ public:
     {
     }
 
+    Property(const Property&) = default;
+
     Property& operator=(const Property& rhs)
     {
         d = rhs.d;
@@ -474,6 +480,8 @@ public:
     {
     }
 
+    Signal(const Signal&) = default;
+
     Signal& operator=(const Signal& rhs)
     {
         d = rhs.d;
