diff --git a/drivers/hwbfm/main/bfmr_main.c b/drivers/hwbfm/main/bfmr_main.c
index 92d90c06..069d080b 100644
--- a/drivers/hwbfm/main/bfmr_main.c
+++ b/drivers/hwbfm/main/bfmr_main.c
@@ -66,7 +66,7 @@ static int __init early_parse_bfmr_enable_flag(char *p)
     {
         if (0 == strncmp(p, "1", strlen("1")))
         {
-            s_is_bfmr_enabled = 1;
+            s_is_bfmr_enabled = 0;
         }
         else
         {
