From e5068aa506cb038feaf441a4c59ef1caddd24b75 Mon Sep 17 00:00:00 2001
From: Brandon Boese <brandonboese@protonmail.com>
Date: Sun, 15 Oct 2023 20:57:38 -0400
Subject: [PATCH] upgrade to C++17

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6267ce4..45a0635 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,7 +76,7 @@ endfunction()
 
 enable_testing()
 
-add_definitions(-std=c++11)
+add_definitions(-std=c++17)
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -Wall")
 # TODO remove the need for -Wno-unused-variable
-- 
2.42.0

