From 93794f028accf15d4e088a9d8c734980774048c3 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Sat, 25 Apr 2020 13:33:53 +0200
Subject: [PATCH 1/5] Replace deprecated g_settings_list_keys usage

Fixes #4
---
 src/qgsettings.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/qgsettings.cpp b/src/qgsettings.cpp
index 693fa1b..0859160 100644
--- a/src/qgsettings.cpp
+++ b/src/qgsettings.cpp
@@ -105,7 +105,7 @@ bool QGSettings::trySet(const QString &key, const QVariant &value)
 QStringList QGSettings::keys() const
 {
     QStringList list;
-    gchar **keys = g_settings_list_keys(priv->settings);
+    gchar **keys = g_settings_schema_list_keys(priv->schema);
     for (int i = 0; keys[i]; i++)
         list.append(qtify_name(keys[i]));
 
-- 
2.26.2

