From 0658a40ba4ef5bd7221642fc051983a53026ca35 Mon Sep 17 00:00:00 2001
From: Marius Gripsgard <marius@ubports.com>
Date: Sat, 28 Dec 2019 17:26:00 +0100
Subject: [PATCH] Add Mir::AttachedState

---
 include/unity/shell/application/Mir.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/unity/shell/application/Mir.h b/include/unity/shell/application/Mir.h
index 04e81d3..efb7b5c 100644
--- a/include/unity/shell/application/Mir.h
+++ b/include/unity/shell/application/Mir.h
@@ -76,6 +76,7 @@ public:
         MaximizedBottomLeftState,
         MaximizedBottomRightState,
         HiddenState,
+        AttachedState,
     };
     Q_ENUM(State)
 
-- 
2.25.2

