diff --git a/src/ubuntu/application/common/application/url_dispatcher/CMakeLists.txt b/src/ubuntu/application/common/application/url_dispatcher/CMakeLists.txt
index 42ff4bd..b765b45 100644
--- a/src/ubuntu/application/common/application/url_dispatcher/CMakeLists.txt
+++ b/src/ubuntu/application/common/application/url_dispatcher/CMakeLists.txt
@@ -1,5 +1,5 @@
 find_package(PkgConfig)
-pkg_check_modules(LIBURL_DISPATCHER REQUIRED url-dispatcher-1)
+pkg_check_modules(LIBURL_DISPATCHER REQUIRED lomiri-url-dispatcher)
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --std=c++11 -fPIC")
 
diff --git a/src/ubuntu/application/common/application/url_dispatcher/session.cpp b/src/ubuntu/application/common/application/url_dispatcher/session.cpp
index 929f029..676f24d 100644
--- a/src/ubuntu/application/common/application/url_dispatcher/session.cpp
+++ b/src/ubuntu/application/common/application/url_dispatcher/session.cpp
@@ -17,7 +17,7 @@
 
 #include "ubuntu/application/url_dispatcher/session.h"
 
-#include <url-dispatcher.h>
+#include <lomiri-url-dispatcher.h>
 
 #include <stdlib.h>
 
@@ -47,5 +47,5 @@ ua_url_dispatcher_session_open(
     DispatchCallBackData *data = (DispatchCallBackData*)malloc(sizeof(DispatchCallBackData));
     data->handler = handler;
     data->user_context = user_context;
-    url_dispatch_send(url, dispatch_callback, data);
+    lomiri_url_dispatch_send(url, dispatch_callback, data);
 }
