diff --git a/docs/qml/CMakeLists.txt b/docs/qml/CMakeLists.txt
index 93e187cc..870e2484 100644
--- a/docs/qml/CMakeLists.txt
+++ b/docs/qml/CMakeLists.txt
@@ -1,9 +1,9 @@
 # add a target to generate API documentation with qdoc
-FIND_PROGRAM(QDOC_EXECUTABLE qdoc)
+FIND_PROGRAM(QDOC_EXECUTABLE qdoc-qt5)
 if(QDOC_EXECUTABLE)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/lomiri-download-manager.qdocconf.in ${CMAKE_CURRENT_BINARY_DIR}/lomiri-download-manager.qdocconf @ONLY)
 add_custom_target(qmldoc
-${QDOC_EXECUTABLE} -qt=qt5 ${CMAKE_CURRENT_BINARY_DIR}/lomiri-download-manager.qdocconf
+${QDOC_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/lomiri-download-manager.qdocconf
 WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
 COMMENT "Generating QML API documentation with qdoc" VERBATIM
 )
