From f8f4b75515367beecff8476fb9418f3a39769237 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Wed, 6 May 2020 20:06:06 +0200
Subject: [PATCH 3/6] UAL3

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e02c01218..c894b20ce 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -81,7 +81,7 @@ pkg_check_modules(GLIB REQUIRED glib-2.0>=2.32)
 pkg_check_modules(LAUNCHER_API REQUIRED unity-shell-launcher=13)
 pkg_check_modules(QMENUMODEL REQUIRED qmenumodel)
 pkg_check_modules(GD3 REQUIRED gnome-desktop-3.0)
-pkg_check_modules(UAL REQUIRED ubuntu-app-launch-2)
+pkg_check_modules(UAL REQUIRED ubuntu-app-launch-3)
 pkg_check_modules(UBUNTUGESTURES REQUIRED UbuntuGestures)
 pkg_check_modules(QTDBUSTEST REQUIRED
     libqtdbustest-1
-- 
2.26.2

