From 8aa82c40a21bd967f73d7a3344f50636a5a312f4 Mon Sep 17 00:00:00 2001
From: Evangelos Ribeiro Tzaras <devrtz@fortysixandtwo.eu>
Date: Wed, 8 Jul 2020 18:45:08 +0200
Subject: [PATCH 5/6] phosh-wayland.c: Indent

---
 src/phosh-wayland.c | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/src/phosh-wayland.c b/src/phosh-wayland.c
index 427625c..ed11341 100644
--- a/src/phosh-wayland.c
+++ b/src/phosh-wayland.c
@@ -67,17 +67,17 @@ registry_handle_global (void *data,
   struct wl_output *output;
 
   if (!strcmp (interface, "phosh_private")) {
-      priv->phosh_private = wl_registry_bind (
-        registry,
-        name,
-        &phosh_private_interface,
-        MIN(4, version));
-  } else  if (!strcmp (interface, zwlr_layer_shell_v1_interface.name)) {
-      priv->layer_shell = wl_registry_bind (
-        registry,
-        name,
-        &zwlr_layer_shell_v1_interface,
-        1);
+    priv->phosh_private = wl_registry_bind (
+      registry,
+      name,
+      &phosh_private_interface,
+      MIN(4, version));
+  } else if (!strcmp (interface, zwlr_layer_shell_v1_interface.name)) {
+    priv->layer_shell = wl_registry_bind (
+      registry,
+      name,
+      &zwlr_layer_shell_v1_interface,
+      1);
   } else if (!strcmp (interface, "wl_output")) {
     output = wl_registry_bind (
       registry,
-- 
2.27.0

