diff --git a/plugins/CMakeLists.txt b/plugins/CMakeLists.txt
index 4d53e21bd..09ac97dae 100644
--- a/plugins/CMakeLists.txt
+++ b/plugins/CMakeLists.txt
@@ -16,7 +16,7 @@ add_subdirectory(Cursor)
 add_subdirectory(GlobalShortcut)
 add_subdirectory(Greeter)
 add_subdirectory(LightDM)
-add_subdirectory(Lights)
+#add_subdirectory(Lights)
 add_subdirectory(Powerd)
 add_subdirectory(ScreenshotDirectory)
 add_subdirectory(SessionBroadcast)
