From 838a57d3f5afa46107ca0e60d1e0324d4010f41a Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Wed, 7 Jun 2023 10:13:15 +0200
Subject: [PATCH] Add missing headers for GCC13

---
 include/qtmir/miral/edid.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/qtmir/miral/edid.h b/include/qtmir/miral/edid.h
index 0845479a..d8f4e369 100644
--- a/include/qtmir/miral/edid.h
+++ b/include/qtmir/miral/edid.h
@@ -19,6 +19,7 @@
 #ifndef MIRAL_EDID_H
 #define MIRAL_EDID_H
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
-- 
2.41.0

