diff --git a/src/platforms/mirserver/displayconfigurationpolicy.cpp b/src/platforms/mirserver/displayconfigurationpolicy.cpp
index 2071bcb3..12055e4c 100644
--- a/src/platforms/mirserver/displayconfigurationpolicy.cpp
+++ b/src/platforms/mirserver/displayconfigurationpolicy.cpp
@@ -83,7 +83,7 @@ void DisplayConfigurationPolicy::apply_to(mg::DisplayConfiguration& conf)
             if (output.connected) {
                 screenCount++;
 
-                if (output.type == mg::DisplayConfigurationOutputType::lvds) {
+                if (output.type == mg::DisplayConfigurationOutputType::dsi) {
                     phoneDetected = true;
                 }
             }
@@ -100,10 +100,10 @@ void DisplayConfigurationPolicy::apply_to(mg::DisplayConfiguration& conf)
             nextTopLeftPosition += output.modes[output.current_mode_index].size.width.as_int();
 
             if (phoneDetected) {
-                if (screenCount == 1 || output.type == mg::DisplayConfigurationOutputType::lvds) {
+                if (screenCount == 1 || output.type == mg::DisplayConfigurationOutputType::dsi) {
                     output.form_factor = mir_form_factor_phone;
                     output.scale = d->m_defaultScale;
-                } else { // screenCount > 1 && output.type != lvds
+                } else { // screenCount > 1 && output.type != dsi
                     output.form_factor = mir_form_factor_monitor;
                     output.scale = 1;
                 }
