From 24351ad41224fdbf2a42dbda6c380a54e4631ab5 Mon Sep 17 00:00:00 2001
From: Marius Gripsgard <marius@ubports.com>
Date: Sun, 8 Sep 2019 19:54:55 +0200
Subject: [PATCH] Wayland hack, set maliit-server to inputmethod type (WIP)

---
 src/platforms/mirserver/wrappedwindowmanagementpolicy.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/platforms/mirserver/wrappedwindowmanagementpolicy.cpp b/src/platforms/mirserver/wrappedwindowmanagementpolicy.cpp
index 3f7ee2ae..8932b244 100644
--- a/src/platforms/mirserver/wrappedwindowmanagementpolicy.cpp
+++ b/src/platforms/mirserver/wrappedwindowmanagementpolicy.cpp
@@ -129,6 +129,12 @@ namespace qtmir
             if (initialSize.isValid() && surfaceType == mir_surface_type_normal) {
                 parameters.size() = toMirSize(initialSize);
             }
+
+            auto surfaceName = requestParameters.name().is_set() ? requestParameters.name().value() : "";
+
+            if (surfaceName == "maliit-server") {
+                parameters.type() = mir_window_type_inputmethod;
+            }
         }
 
         parameters.userdata() = std::make_shared<ExtraWindowInfo>();
-- 
2.39.1

