From 4854d238195f1ecf80f232ed6072c20d6f3d8716 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Sun, 15 Aug 2021 16:52:09 +0200
Subject: [PATCH 5/5] fixups

---
 qml/Panel/Indicators/IndicatorsLight.qml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qml/Panel/Indicators/IndicatorsLight.qml b/qml/Panel/Indicators/IndicatorsLight.qml
index 3ffab622e..681f0d8ee 100644
--- a/qml/Panel/Indicators/IndicatorsLight.qml
+++ b/qml/Panel/Indicators/IndicatorsLight.qml
@@ -211,7 +211,7 @@ console.log("no support for Multicolor LED. " + indicatorState)
         }
     }
 
-    Component.onCompleted: Leds.state = Lights.Off
+    Component.onCompleted: Leds.state = Leds.Off
     Component.onDestruction: Leds.state = Leds.Off
 
     property var _colorBinding: Binding {
-- 
2.39.1

