From 27cc8bf8d3d46f1aa04e39a81c1252c75d9e3c41 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Sun, 15 Aug 2021 16:52:09 +0200
Subject: [PATCH 06/11] Use correct value for Leds.state

---
 qml/Panel/Indicators/IndicatorsLight.qml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qml/Panel/Indicators/IndicatorsLight.qml b/qml/Panel/Indicators/IndicatorsLight.qml
index 3ffab622e..681f0d8ee 100644
--- a/qml/Panel/Indicators/IndicatorsLight.qml
+++ b/qml/Panel/Indicators/IndicatorsLight.qml
@@ -211,7 +211,7 @@ console.log("no support for Multicolor LED. " + indicatorState)
         }
     }
 
-    Component.onCompleted: Leds.state = Lights.Off
+    Component.onCompleted: Leds.state = Leds.Off
     Component.onDestruction: Leds.state = Leds.Off
 
     property var _colorBinding: Binding {
-- 
2.43.0

