From 34793a091ce7b0c3c445ad530b37da6723429111 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Sat, 15 Jul 2023 11:36:12 +0300
Subject: [PATCH] Add more missing headers

---
 src/core/state_machine.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/state_machine.h b/src/core/state_machine.h
index 59e1cd4..faba259 100644
--- a/src/core/state_machine.h
+++ b/src/core/state_machine.h
@@ -23,6 +23,7 @@
 #include "power_supply.h"
 
 #include <chrono>
+#include <string>
 
 namespace repowerd
 {
-- 
2.41.0

