From 53bf545a52c20c79924eee896d95de5255e23888 Mon Sep 17 00:00:00 2001
From: Antoine Fontaine <antoine.fontaine@epfl.ch>
Date: Tue, 23 Jun 2020 12:13:08 +0200
Subject: [PATCH 1/3] layer_shell: arrange layers in reverse order

---
 sway/desktop/layer_shell.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sway/desktop/layer_shell.c b/sway/desktop/layer_shell.c
index 60a8f06b..f9cfea82 100644
--- a/sway/desktop/layer_shell.c
+++ b/sway/desktop/layer_shell.c
@@ -96,7 +96,7 @@ static void arrange_layer(struct sway_output *output, struct wl_list *list,
 	struct wlr_box full_area = { 0 };
 	wlr_output_effective_resolution(output->wlr_output,
 			&full_area.width, &full_area.height);
-	wl_list_for_each(sway_layer, list, link) {
+	wl_list_for_each_reverse(sway_layer, list, link) {
 		struct wlr_layer_surface_v1 *layer = sway_layer->layer_surface;
 		struct wlr_layer_surface_v1_state *state = &layer->current;
 		if (exclusive != (state->exclusive_zone > 0)) {
-- 
2.28.0

