From 28dffecefecc4a7b5cdb584581133fed0df2b20c Mon Sep 17 00:00:00 2001
From: Jan Arne Petersen <janarne@gmail.com>
Date: Tue, 18 Apr 2017 00:20:25 +0200
Subject: [PATCH] Add a subview to make keyboard work

---
 src/plugin/inputmethod.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/plugin/inputmethod.cpp b/src/plugin/inputmethod.cpp
index 63d1e543..49d1ec09 100644
--- a/src/plugin/inputmethod.cpp
+++ b/src/plugin/inputmethod.cpp
@@ -208,7 +208,10 @@ QList<MAbstractInputMethod::MInputMethodSubView>
 InputMethod::subViews(Maliit::HandlerState state) const
 {
     Q_UNUSED(state)
+    Q_D(const InputMethod);
+
     QList<MInputMethodSubView> views;
+    views << MInputMethodSubView{ d->activeLanguage, d->activeLanguage };
     return views;
 }
 
-- 
2.39.0

