From d58022eb731523015870388a6a9bcbc656a8b8a5 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Thu, 5 Mar 2020 13:36:06 +0100
Subject: [PATCH 1/7] Don't hardcode Debian-specific python3 path

---
 tests/tests.pro | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tests/tests.pro b/tests/tests.pro
index bdd77a1c1..796029f82 100644
--- a/tests/tests.pro
+++ b/tests/tests.pro
@@ -1,7 +1,9 @@
 TEMPLATE = subdirs
 SUBDIRS += unit autopilot
 
-autopilot_module.path = $$[QT_INSTALL_PREFIX]/lib/python3/dist-packages/ubuntuuitoolkit
+PYTHONDIR = $$system(python3 -c \"from distutils.sysconfig import get_python_lib; print(get_python_lib())\")
+
+autopilot_module.path = $$PYTHONDIR/ubuntuuitoolkit
 autopilot_module.files = autopilot/ubuntuuitoolkit/*
 
 SUBDIRS += api
-- 
2.28.0

