From 7f431fa134dd757d6d238a2d3c124051e52258e8 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Tue, 15 Sep 2020 17:48:47 +0200
Subject: [PATCH 4/7] Don't build app-launch-profiler

---
 ubuntu-sdk.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ubuntu-sdk.pro b/ubuntu-sdk.pro
index 716c3055e..647b53ad1 100644
--- a/ubuntu-sdk.pro
+++ b/ubuntu-sdk.pro
@@ -11,7 +11,7 @@ load(qt_parts)
 src_uitk_launcher.subdir = ubuntu-ui-toolkit-launcher
 src_uitk_launcher.depends = sub-src
 
-SUBDIRS += po app-launch-profiler src_uitk_launcher apicheck
+SUBDIRS += po src_uitk_launcher apicheck
 !CONFIG(no_docs) {
     SUBDIRS += documentation
 }
-- 
2.28.0

